pkg/k8sutil/services.go

```diff
@@ -33,40 +38,66 @@ func (e IngressSetupPendingError) Error() string {
 func GetServiceEndpointIPs(service corev1.Service) ([]string, bool, error) {
 	ips := make([]string, 0)
 
-	var hasHostname bool
+	var hostname string
+
+	if v, ok := service.GetAnnotations()[ServiceHostnameOverrideAnnotation]; ok && v != "" {
```

Comment:
Could `service.GetAnnotations()` return `nil`? Because in that case it should be handled too.